onnx/defs/text/defs.cc

```diff
@@ -5,6 +5,44 @@
 #include "onnx/defs/schema.h"
 
 namespace ONNX_NAMESPACE {
+static const char* ParseDateTime_doc = R"DOC(Parse a datetime string into a (floating point) Unix time stamp.)DOC";
+ONNX_OPERATOR_SET_SCHEMA(
+    ParseDateTime,
+    20,
+    OpSchema()
+        .Input(0, "X", "Tensor with datetime strings", "T1", OpSchema::Single, true, 1, OpSchema::NonDifferentiable)
+        .Output(0, "y", "Unix time stamps", "T2", OpSchema::Single, true, 1, OpSchema::NonDifferentiable)
+        .Attr("format", "Format description in the syntax of C's `strptime`.", AttributeProto::STRING)
+        .Attr(
+            "unit",
+            "Unit of the returned time stamp. Allowed values are: 's' (second), 'ms' (millisecond), 'us' (microsecond) or 'ns' (nanosecond).",
+            AttributeProto::STRING)
+        .Attr(
+            "default",
+            "Default value to be used if the parsing fails. The tensor must be of rank 0 and either of type `tensor(int64)` or `tensor(double)`. The tensor type is the output type. If 'default' is specified, the output type is `tensor(int64)` and the behavior for failing to parse an input element is implementation defined.",
+            AttributeProto::TENSOR,
+            OPTIONAL_VALUE)
+
+        .TypeConstraint("T1", {"tensor(string)"}, "UTF-8 datetime strings")
+        .TypeConstraint("T2", {"tensor(double)", "tensor(int64)"}, "Output type depends on 'default' attribute.")
```

Comment:
Do we need to support double here? In principle, users can encode it as `Where( Equal(value, default), NaN, Cast(value, float))`.  Specifically, if we are not worried about nano-seconds (which strptime doesn't seem to support anyway), is the dynamic range of int64 not sufficient (sticking to microseconds)?